mvrx/src/main/kotlin/com/airbnb/mvrx/lifecycleAwareLazy.kt

```diff
@@ -29,9 +34,18 @@ class lifecycleAwareLazy<out T>(private val owner: LifecycleOwner, initializer:
     private val lock = this
 
     init {
-        // owner.lifecycle.addObserver must be invoked on main thread otherwise addObserver will throw an IllegalStateException.
-        // createUnsafe disables the main thread check.
-        LifecycleRegistry.createUnsafe(owner).addObserver(object : DefaultLifecycleObserver {
+        if (isMainThread()) {
+            initializeOnCreatedOrGreater(owner)
+        } else {
+            Handler(Looper.getMainLooper()).post {
+                if (owner.lifecycle.currentState < Lifecycle.State.CREATED || isInitialized()) return@post
```

Comment:
@marukami this currently returns and does not initialize if the lifecycle is at the `INITIALIZED` state, which seems like it could be a problem. afaict we would only want to cancel if the lifecycle is destroyed

```suggestion
                if (owner.lifecycle.currentState == DESTROYED || isInitialized()) return@post
```

Reply:
Ah, yep. I follow what you mean here.